core/Pos/Binary/Class/Core.hs

```diff
@@ -1,171 +1,586 @@
-{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE PolyKinds           #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE TypeOperators       #-}
+module Pos.Binary.Class.Core
+    ( Bi(..)
+    , encodeBinary
+    , decodeBinary
+    , enforceSize
+    , matchSize
+    -- * CBOR re-exports
+    , encodeListLen
+    , decodeListLen
+    , decodeListLenOf
+    , Encoding
+    , Decoder
+    -- * GHC-Generics-based encoding & decoding
+    , genericEncode
+    , genericDecode
+    ) where
 
-#include "MachDeps.h"
+import           Codec.CBOR.Decoding
+import           Codec.CBOR.Encoding
+import           GHC.Generics
```

Comment:
`defaultEncodeList` vs `encodeList` is not about naming, these are two different functions (`encodeList` is from `Bi` class, while `defaultEncodeList` is default implementation which can be different for arbitrary `a`).